### DISCUSSION TITLE
Render screenshots in different formats

### DISCUSSION THREAD

**Discussion by NEONFIVE** [NVlabs/instant-ngp#301] [Q&A] [answered]
Hi, i am just wondering if there is an optional way to render and write the screenshots in a format like 32bit .exr file. 
For the depthmap it would be nice to have such an uncompressed format to further use in addional software like Blender etc.

Any information on that?

**Comment by Tom94** [comment, marked as answer]
You can use the python bindings for this. `testbed.render` return a numpy array in linear colors, so you can write an exr file as easily as:
```python
common.write_image("screenshot.exr", testbed.render(1920, 1080, spp=1, linear=True))
```

**Reply by NEONFIVE** [reply]
thanks for the fast reply. but it raised an error (path cleared with "_path"):

rendering _path\r_003.exr
Traceback (most recent call last):
  File "_path\run.py", line 312, in <module>
    write_image(outname, image)
  File "_path\common.py", line 188, in write_image
    write_image_imageio(file, img, quality)
  File "_path\common.py", line 130, in write_image_imageio
    imageio.imwrite(img_file, img, **kwargs)
  File "_path\Python39\site-packages\imageio\v2.py", line 195, in imwrite
    with imopen(uri, "wi", plugin=format) as file:
  File "_path\Python\Python39\site-packages\imageio\core\imopen.py", line 333, in imopen
    raise err_type(err_msg)
ValueError: Could not find a backend to open `_path/screenshots\r_003.exr`` with iomode `wi`.


Any help on that?

**Reply by Tom94** [reply]
This looks like an error in the Python `imageio` library. Google tells me you might need to install a plugin to be able to write EXR files using imageio; see https://imageio.readthedocs.io/en/v2.4.1/format_exr-fi.html#exr-fi

**Reply by NEONFIVE** [reply]
Hi Tom, thank you very much, it works now. had to install the plugin.

**Reply by felixgeen** [reply]
i used the command "imageio_download_bin freeimage" to install the plugin. could you explain how to make sure it is installed? I have the freeimage-3.15.1-win64.dll file downloaded and sitting in its default location (\AppData\Local\imageio\freeimage) but I still get the same error as you had NEONFIVE. 

Does the DLL need to be in a specific folder to work?

**Reply by felixgeen** [reply]
`conda install -c conda-forge freeimage` worked for me
